### ISSUE TITLE
Export causes huge file sizes

### ISSUE THREAD

**Issue by wolfgangholzer** [thundernest/import-export-tools-ng#334]
When exporting selected mails via right-click, the result is a saved file size up to 25 times (!) the original file size.

Example:
When moving the mail to an open folder with the mouse, the resulting size is 300 KB.
When saving via ImportExport NG, the file size is 7.700 KB!
You can find both files attached.

Can you help?

**Comment by cleidigh** [comment]
@wolfgangholzer 
I will do my best...
Some extra info:
- OS, TB and IETNG versions 
- What export operation were you doing 
- have you tried similar exports for comparison 
- when did this start
- anything changed when this happened 
- I don't see any files 

@cleidigh

**Comment by wolfgangholzer** [comment]
Good morning/Guten Morgen!

OS: Windows 11 Home, 21H2
TB: 91.11.0 (64-Bit)
IETNG: 12.0.1 : Maintenance Release - July 8, 2022

Exporting selected mails via mouse right-click, saving in EML-format.
Tried with a single file, with 3-10, with many files. 
All sizes increased.
Started approx. a month ago.
No changes noted.

Files are here:
[https://we.tl/t-yFowWlIJkf ](https://we.tl/t-yFowWlIJkf)

This also happens independently of original file sizes.
E.g. a mail originally sized 6 MB results in 16 MB.

Best regards,
Wolfgang Holzer

**Comment by wolfgangholzer** [comment]
Maybe my message dated 14.07. didn't reach you... let's try it this way.

How did you get the first/single email file?
I just moved it from my Thunderbird inbox to an open explorer folder with the mouse (drag + drop).

Where did you export from, imap, pop or local?
It's an IMAP mail account, but I exported from the local TB inbox folder.

What is the source of these emails, was any import involved?
No import. The messages are just like they came in on my office account, sent by project partners.

All eml exports from anywhere produce the "extended" files, yes?
Yes. Not all of them with the same multipier, but all are enlargened significantely.

Would you mind if I send you those?
No, of course not. Thanks for your support.

Best regards,
Wolfgang Holzer

**Comment by cleidigh** [comment]
@wolfgangholzer 
Did you receive my test emails and comments?
I sent from test1@kokkini.net
@cleidigh

**Comment by wolfgangholzer** [comment]
No, sorry, I did not receive any files.
Can yyou please send them to this address: wolfgang@holzer.co

**Comment by cleidigh** [comment]
@wolfgangholzer 
I just re-sent the two emails.
So try the first exporting from inbox then from a local folder.
I also wonder what viewing the message source  Ctrl-U looks like.
@cleidigh

**Comment by cleidigh** [comment]
@wolfgangholzer 
Just had a spam bounce from your server.
Perhaps if you email me it will mark my address as good?
test1@kokkini.net

**Comment by cleidigh** [comment]
@wolfgangholzer 
ok i got no bounces so you should have our two test messages.
Let's compare exporting from inbox and a new local folder.
Q: Have you done any compacting or folder repair?
I ask because your scenario so far eludes some explanation. IETNG exports what TB gives it for a given message.
I'm wondering if the profile is somehow corrupted.

@cleidigh

**Comment by wolfgangholzer** [comment]
All files as explained in my email:

I moved mail 1 to a local folder, called "CL (Test TB).
[Test email 1 & 2.zip](https://github.com/thundernest/import-export-tools-ng/files/9266281/Test.email.1.2.zip)

The exported it to mbox, html, eml.

Mail 2 has been exported directly from the inbox.

All mails seem to stay very small.

**Comment by cleidigh** [comment]
@wolfgangholzer 
I believe you are correct, there is nothing rythmic or mathematical about the issue.
I am not sure if you tested single emails from inbox?
As I think I said before I am beginning to think it's a corrupt database or msf files.
That said I always want to proceed with caution.
Have you done "folder repair" under folder/properties?
This may fix some issues, but I would want you to have a backup first.
One more experiment, what happens if you copy a set of emails to a temp folder under inbox then export from there.
This new folder will get a new msf file which if the export goes well would be good indirect affirmation that the inbox msf is corrupt.
@cleidigh

**Comment by wolfgangholzer** [comment]
Good morning Christopher! (yes, this is Europe :-)
Testmail 2 has been exported directly from inbox.
I did not start folder repair yet.
Now, I copied all TB data files to have a backup. Then restarted TB, chose "folder repair" for my main folder, and one of the local folders. 
After this, the export from the local folder resulted in CORRECT file sizes for approx. 50 mails!

At last, I copied 10 mails to a new folder under inbox and exported them from there in eml format.
Unfortunately, this again resulted in massively enlarged eml-files, although not for all mails. But some of them had been 20 KB and grew to 25.000 KB (25 MB), another from 35 KB to 22.000 KB!

Then, I retried the local folder from above. Again, all file sizes are correct.

Does this give you any clue?

**Comment by cleidigh** [comment]
Wolfgang, Guten morgen
Some Progress!
Ok repairing the local folder worked, but I was not clear on inbox. Did you repair inbox?
The total approach here would be to delete all msf files in the profile tree while TB is shutdown then restart and you can compact the folders.
I am hoping this works.
@cleidigh

**Comment by cleidigh** [comment]
@wolfgangholzer 
Just checking if you have resolved this?
@cleidigh

**Comment by wolfgangholzer** [comment]
Dear Christopher,
I found a workaround, so the issue can be avoided.
It appears, that this size increase happens when exporting just marked mails from any folder. So I created an "handling folder", move the mails to be exported there, and then export all of them at once.
This results in the right size for all mails...
I think we can close this conversation, and I'd like to thank you for your support.

**Comment by cleidigh** [comment]
@wolfgangholzer 
I must apologize because I kind of disappeared on you.
That is good you found a workaround.
I still feel like there is some db issue since IETNG does not control the message boundaries.
keep an eye open for anything strange.
@cleidigh

**Comment by Mthbzr** [comment]
@cleidigh,
I agree with your last comment : "there is some db issue since IETNG does not control the message boundaries".
I compared two eml files : one mail exported with IETNG and the same mail exported with "save as" in TB.
The file exported with IETNG has the same content than the file exported with "save as" and a lot of lines with other informations (other senders, dates, ...) wich give a huge file. Once this lines are deleted, the files are the same size.
